### PR TITLE
feat(addOrder): signature verification from smartcontract

### PR DESCRIPTION
**Pull request by Mitsichury**
#11

### CODE REVIEW

File: src/client/Web3SwapClient.ts [hunk 1]
```diff
@@ -51,6 +51,23 @@ export class Web3SwapClient {
...
     }
 
+    public async isValidOrder(dbOrder: DbOrder) {
+        let isValid = false;
+        const contract = this.contracts[dbOrder.chainId];
+        if (!contract) {
+            return Promise.resolve(isValid);
+        }
+        try {
+            isValid = await contract.check(
```
Comment (piersss) [inline src/client/Web3SwapClient.ts:61]: `check` returns `[BigNumber, string[]]`. So you need to check if BigNumber is 0. Or use the `checkResultToErrors` from @airswap/utils.

The way I implemented it in marketplace:

https://github.com/airswap/airswap-marketplace/blob/8804e85c45aff27ed0a7dbf9d7b968531eda3d97/src/redux/stores/orders/ordersApi.ts#L103

So in your code I would do it like this:

```
const [count, checkErrors] = await Swap.getContract(provider.getSigner(), chainId).check(senderWallet, order);
const errors = (count && checkErrors) ? checkResultToErrors(checkErrors, count) as SwapError[] : [];

return !!errors.length
```

File: src/client/Web3SwapERC20Client.ts [hunk 2]
```diff
@@ -52,6 +51,32 @@ export class Web3SwapERC20Client {
...
     }
 
+    public async isValidOrder(dbOrder: DbOrderERC20) {
+        let isValid = false;
+        const contract = this.contracts[dbOrder.chainId];
+        if (!contract) {
+            return Promise.resolve(isValid);
+        }
+        try {
+            isValid = await contract.check(
```
Comment (piersss) [inline src/client/Web3SwapERC20Client.ts:61]: Same as above.

File: src/client/Web3SwapClient.ts (hunk 1 above)
Comment (Mitsichury) [reply src/client/Web3SwapClient.ts]: Fixed thank you :D

File: src/client/Web3SwapClient.ts [hunk 3]
```diff
@@ -51,7 +52,32 @@ export class Web3SwapClient {
...
+                fullOrder.sender.wallet,
+                fullOrder
+            )
+            const errors = checkResultToErrors(response[1], response[0])
+            for (let index in errors) {
+                if(Object.keys(OrderErrors).includes(errors[index])) {
+                    isValid = false;
+                    break;
+                }                
+            }
```
Comment (piersss) [inline src/client/Web3SwapClient.ts:72]:
```suggestion
isValid = !errors.some(error => Object.keys(OrderErrors).includes(error));
```

File: src/client/Web3SwapERC20Client.ts [hunk 4]
```diff
@@ -52,7 +52,41 @@ export class Web3SwapERC20Client {
...
+                dbOrder.r,
+                dbOrder.s
+            )
+            const errors = checkResultToErrors(response[0], response[1])
+            for (let index in errors) {
+                if(Object.keys(ERC20Errors).includes(errors[index])) {
+                    isValid = false;
+                    break;
+                }                
+            }
```
Comment (piersss) [inline src/client/Web3SwapERC20Client.ts:81]:
```suggestion
            isValid = !errors.some(error => Object.keys(ERC20Errors).includes(error));
```

File: src/client/Web3SwapClient.ts (hunk 3 above)
Comment (piersss) [reply src/client/Web3SwapClient.ts:72]: `some` is an array function that returns true if condition is met, then it breaks, some behaviour as your for loop.

Comment (Mitsichury) [reply src/client/Web3SwapClient.ts]: Ho thank you this is much better :D !

File: src/client/Web3SwapERC20Client.ts (hunk 4 above)
Comment (Mitsichury) [reply src/client/Web3SwapERC20Client.ts]: Changed :)

**Review by piersss — APPROVED** [review]
LGTM

**Pull request merged**